corehq/apps/app_manager/util.py

```diff
@@ -507,84 +508,102 @@ def get_sort_and_sort_only_columns(detail, sort_elements):
+                                                   include_groups=True, include_translations=True)
+                for compiled_question in self._compile_form_questions(form.unique_id, question)
+            ]
+        except XFormException as exception:
+            form_meta['questions'] = []
+            form_meta['error'] = {
+                'details': six.text_type(exception),
+                'edit_url': reverse('form_source', args=[self.domain, self.app._id, form.unique_id]),
             }
-            try:
-                questions = form.get_questions(
-                    app.langs,
-                    include_triggers=True,
-                    include_groups=True,
-                    include_translations=True
-                )
-                seen_save_to_case = set()
-                for question in questions:
-                    is_save_to_case = '/case/' in question['value']
-                    response = FormQuestionResponse(question).to_json()
-                    response['load_properties'] = case_meta.get_load_properties(form.unique_id, question['value'])
-                    response['save_properties'] = case_meta.get_save_properties(form.unique_id, question['value'])
-                    if is_save_to_case:
-                        response['type'] = 'SaveToCase'
-                    form_meta['questions'].append(response)
-
-                    if is_save_to_case:
-                        question_path = question['value'].split('/case/')[0]
-                        if question_path not in seen_save_to_case:
-                            response = FormQuestionResponse(question).to_json()
-                            response['load_properties'] = case_meta.get_load_properties(form.unique_id, question_path)
-                            response['save_properties'] = case_meta.get_save_properties(form.unique_id, question_path)
-                            response['value'] = question_path
-                            response['label'] = response['label'].split('/case/')[0]
-                            response['type'] = 'SaveToCase'
-                            form_meta['questions'].append(response)
-                        seen_save_to_case.add(question_path)
-
-
-
-
-            except XFormException as e:
-                form_meta['error'] = {
-                    'details': six.text_type(e),
-                    'edit_url': reverse(
-                        'form_source',
-                        args=[domain, app._id, form.unique_id]
-                    ),
-                }
-                form_meta['module'] = copy(module_meta)
-                errors.append(form_meta)
-            else:
-                forms.append(form_meta)
+            self.errors.append(form_meta)
+        return form_meta
+
+    def _compile_form_questions(self, form_unique_id, question):
+        from corehq.apps.reports.formdetails.readable import FormQuestionResponse
```

Comment:
nit: is there a reason this can't be moved to the top of the file? (same in `_get_save_to_case_root_node`)

Reply:
Good point. This was causing an import loop, which suggests to me that this was in the wrong place. I've moved this whole class alongside the other code that does the same things here: https://github.com/dimagi/commcare-hq/pull/23409/commits/e55e7832defb68b81e08160b3a7bb94c336fc524